main.go

```diff
@@ -98,27 +102,50 @@ func main() {
 	}
 
 	qfService := web.NewQuickFeedService(logger, db, scms, bh, runner)
-	go web.New(qfService, *public, *httpAddr)
-
-	lis, err := net.Listen("tcp", *grpcAddr)
+	// TODO(vera): harcoded or command line flags?
+	certFile := os.Getenv("CERT")
+	certKey := os.Getenv("CERT_KEY")
+	if certFile == "" || certKey == "" {
+		log.Fatal("CERT or CERT_KEY is not set")
+	}
+	grpcServer, err := web.ServerWithCredentials(logger, certFile, certKey)
 	if err != nil {
-		log.Fatalf("failed to start tcp listener: %v\n", err)
+		log.Fatalf("failed to generate gRPC server credentials: %v\n", err)
 	}
-	opt := grpc.ChainUnaryInterceptor(auth.UserVerifier(), qf.Interceptor(logger))
-	grpcServer := grpc.NewServer(opt)
+
+	qf.RegisterQuickFeedServiceServer(grpcServer, qfService)
+	grpcWebServer := grpcweb.WrapServer(grpcServer)
+
+	multiplexer := web.GrpcMultiplexer{
+		MuxServer: grpcWebServer,
+	}
+
+	// Register HTTP endpoints and webhooks
+	router := qfService.RegisterRouter(authConfig, scms, multiplexer, *public)
+
 	// Create a HTTP server for prometheus.
 	httpServer := &http.Server{
 		Handler: promhttp.HandlerFor(reg, promhttp.HandlerOpts{}),
 		Addr:    fmt.Sprintf("0.0.0.0:%d", 9097),
 	}
 	go func() {
 		if err := httpServer.ListenAndServe(); err != nil {
-			log.Fatal("Unable to start a http server.")
+			log.Fatalf("failed to start a http server: %v\n", err)
```

Comment:
Note that there is a difference between errors that are returned from a function (they should use lower-case initial letter), but when actually logging a message we should use Capitalization. See [here for details](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

Reply:
Right, but the rest of the `log.Fatalf()` messages are not capitalized, this is why I changed this one. Should I fix all the occurrences in the `main.go`?

Reply:
Yes.